### DISCUSSION TITLE
Lifecycle Hook After Response is Sent

### DISCUSSION THREAD

**Discussion by Geometrically** [nuxt/nuxt#18564] [Questions]
Hello,

Is there a nitro/app hook which allows you to hook into after the response has been sent to the client?

Specifically I have an analytics function (which sends a request to a server) which I do not want to block any part of the rendering which executes on the server side. Before, on Nuxt 2, I used the routeDone hook. I am looking for something similar.

**Comment by dulnan** [comment]
I think the closest is `nitroApp.hooks.hook('render:response')`, but that happens _before_ the response is sent. And I think it's also only called when the Nuxt app is rendered (e.g. not when a custom server route response is returned).

But in your case you could just not `await` the analytics request, so the response would be sent immediately, while the analytics request is still pending. I think that should work. You could also buffer these API requests for some time and then periodically execute them at once to prevent having too many of these API requests pending at once.

I did have a scenario though where I had to "catch" all responses sent, from anywhere. I ended up overwriting the `node.res.end()` method to intercept when the response is sent. There's a plugin in nitro itself that demonstrates how this is done:
https://github.com/unjs/nitro/blob/main/src/runtime/timing.ts#L15

**Comment by Baroshem** [comment]
Maybe any of the nuxt hooks could be useful here? 

https://github.com/nuxt/nuxt/blob/main/packages/schema/src/types/hooks.ts#L53
